app/src/main/java/com/epfl/beatlink/viewmodel/spotify/api/SpotifyApiViewModel.kt

```diff
@@ -35,6 +35,30 @@ open class SpotifyApiViewModel(
 
   var currentArtist by mutableStateOf(SpotifyArtist("", "", listOf(), 0))
 
+  fun getPlaylistTracks(
+      playlistID: String,
+      onSuccess: (List<SpotifyTrack>) -> Unit,
+      onFailure: (List<SpotifyTrack>) -> Unit
+  ) {
+    viewModelScope.launch {
+      val result = apiRepository.get("playlists/$playlistID/tracks?limit=50")
```

Comment:
Are you sure about the track limit? The task requirements states "The call should fetch all the tracks of a given playlist id". Maybe you have a good reason but you might wanna reconsider this.

Reply:
The idea is to use this API call to display a preview of the user's playlists on the profile screen. Since this is supposed to be a preview only, an amount of 50 tracks seems reasonably large already. This shouldn't be used anywhere else and we don't plan to fetch the entirety of the tracks of a playlist in the current standing of the app, so this should be fine.

Reply:
Seems legit :)